okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -56,7 +56,12 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
         } else {
           // Write the request body if the "Expect: 100-continue" expectation was met.
           val bufferedRequestBody = exchange.createRequestBody(request, false).buffer()
-          requestBody.writeTo(bufferedRequestBody)
+          try {
+            requestBody.writeTo(bufferedRequestBody)
+          } catch (ioe: IOException) {
+            exchange.noNewExchangesOnConnection()
```

Comment:
Should probably be only called for HTTP/1.1

Reply:
I think `exchange#detachWithViolence` will do what you want. It’ll even wrap the IOException if the cause is suspected to be a timeout.